### PR TITLE
allow combining memoization with step selection

### PR DESCRIPTION
**Pull request by sryza**
The set of steps to execute is the intersection of the unmemoized steps and the selected steps.

It wasn't immediately obvious to me where to add a test for this, but happy to do so if the experts on this code have recommendations on where to add it.

### CODE REVIEW

**Comment by alangenfeld** [comment]
`python_modules/dagster/dagster_tests/execution_tests/test_memoized_dev_loop.py` maybe - I think theres enough scaffolding there to reference to add a test for memoization + step selection

**Review by alangenfeld — CHANGES_REQUESTED** [review]
to your queue for test (or convincing test plan)

File: python_modules/dagster/dagster/core/execution/plan/plan.py [hunk 1]
```diff
@@ -907,8 +906,12 @@ def build_memoized_plan(
                 if not io_manager.has_output(context):
                     unmemoized_step_keys.add(step_output_handle.step_key)
 
+        if selected_step_keys is not None:
+            step_keys_to_execute = list(unmemoized_step_keys & set(selected_step_keys))
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/core/execution/plan/plan.py:910]: nit: maybe a comment here just for extra clarification for dumb dumbs like me

**Review by alangenfeld — APPROVED** [review]
nice ya this looks good to me

File: python_modules/dagster/dagster/core/execution/plan/plan.py (hunk 1 above)
Comment (sryza) [reply python_modules/dagster/dagster/core/execution/plan/plan.py:910]: 👍

**Pull request merged**